src/lib.rs

```diff
@@ -68,7 +29,7 @@ pub type Result<T> = result::Result<T, Error>;
 ///
 /// The function can return an error if the lexer,
 /// the parser or the evaluator encounters an error.
-pub fn decode(encoded_str: &[u8]) -> Result<String> {
+pub fn decode(encoded_str: &[u8]) -> Result<String, Error> {
```

Comment:
Idem `Result`

Reply:
Ah you missed this one!

Reply:
Ah, right. I didn't apply it here because I thought that it might be better to "read" for the user if he/she can directly see what the error-value is gonna be.

Reply:
Yeah true, but I would prefer to stay consistent with the other files. It's either we put all or none, but not half. The `Result` alias was already there before, so I would keep it to minimize the changes :ok_hand:

Reply:
alright, fixed!